config/src/config/transaction_filter_type.rs

```diff
@@ -0,0 +1,192 @@
+///                 - Allow:
+///                     EntryFunction:
+///                         - "0000000000000000000000000000000000000000000000000000000000000001"
+///                         - test
+///                         - new
+///                 - Deny: All
+/// This filter allows transactions from the sender with address f8871acf2c827d40e23b71f6ff2b9accef8dbb17709b88bd9eb95e6bb748c25a or
+/// from the module with address 0000000000000000000000000000000000000000000000000000000000000001 or entry functions
+/// test::check and test::new from the module 0000000000000000000000000000000000000000000000000000000000000001. All other transactions are denied.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct Filter {
+    rules: Vec<Rule>,
+}
+
+impl Filter {
+    pub fn empty() -> Self {
+        Self { rules: vec![] }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.rules.is_empty()
+    }
+
+    pub fn add_deny_all(mut self) -> Self {
+        self.rules.push(Rule::Deny(Matcher::All));
+        self
+    }
+
+    pub fn add_deny_block_id(mut self, block_id: HashValue) -> Self {
+        self.rules.push(Rule::Deny(Matcher::BlockId(block_id)));
+        self
+    }
+
+    pub fn add_deny_transaction_id(mut self, txn_id: HashValue) -> Self {
+        self.rules.push(Rule::Deny(Matcher::TransactionId(txn_id)));
+        self
+    }
+
+    pub fn add_allow_sender(mut self, sender: AccountAddress) -> Self {
+        self.rules.push(Rule::Allow(Matcher::Sender(sender)));
+        self
+    }
+
+    pub fn add_deny_sender(mut self, sender: AccountAddress) -> Self {
+        self.rules.push(Rule::Deny(Matcher::Sender(sender)));
+        self
+    }
+
+    pub fn add_allow_module_address(mut self, address: AccountAddress) -> Self {
+        self.rules
+            .push(Rule::Allow(Matcher::ModuleAddress(address)));
+        self
+    }
+
+    pub fn add_deny_module_address(mut self, address: AccountAddress) -> Self {
+        self.rules.push(Rule::Deny(Matcher::ModuleAddress(address)));
+        self
+    }
+
+    pub fn add_deny_entry_function(
+        mut self,
+        address: AccountAddress,
+        module_name: String,
+        function: String,
+    ) -> Self {
+        self.rules.push(Rule::Deny(Matcher::EntryFunction(
+            address,
+            module_name,
+            function,
+        )));
+        self
+    }
+
+    pub fn add_allow_entry_function(
+        mut self,
+        address: AccountAddress,
+        module_name: String,
+        function: String,
+    ) -> Self {
+        self.rules.push(Rule::Allow(Matcher::EntryFunction(
+            address,
+            module_name,
+            function,
+        )));
+        self
+    }
+
+    pub fn allows(&self, block_id: HashValue, txn: &SignedTransaction) -> bool {
+        for rule in &self.rules {
+            match rule.eval(block_id, txn) {
+                EvalResult::Allow => return true,
```

Comment:
nit: Might be worth adding an comment to this method that the first rule match is what is chosen 😄

Reply:
I have added some comment at the top to make it clear but no harm in adding it again here.